### PR TITLE
[MXNET-1235] Add a test for AdaMax optimizer

### PR DESCRIPTION
**Pull request by D-Roberts**
## Description ##
In issue https://github.com/apache/incubator-mxnet/issues/10923 it was mentioned that some optimizers do not have tests. AdaMax was one of them. One test was added.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- add PyAdamax python reference
- add test_adamax

## Comments ##
- row_sparse cases are not included.

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [Test, pr-awaiting-review]

File: tests/python/unittest/test_optimizer.py [hunk 1]
```diff
@@ -613,6 +612,88 @@ def test_adam():
...
+                        kwarg.update(cg_option)
+                        kwarg.update(rg_option)
+                        kwarg.update(wd_option)
+                        kwarg.update(mp_option)
+                        if (dtype == np.float16 and
+                                ('multi_precision' not in kwarg or
+                                    not kwarg['multi_precision'])):
+                            continue
+                        compare_optimizer(opt1(**kwarg), opt2(**kwarg), shape, dtype,
+                                          rtol=1e-4, atol=2e-5)
```
Comment (anirudhacharya) [inline tests/python/unittest/test_optimizer.py]: why not leave rtol and atol to default values?

File: tests/python/unittest/test_optimizer.py [hunk 2]
```diff
@@ -613,6 +612,88 @@ def test_adam():
...
+@with_seed()
+def test_adamax():
+    opt1 = PyAdamax
+    opt2 = mx.optimizer.Adamax
+    shape = (3, 4, 5)
+    cg_options = [{}, {'clip_gradient': 0.4}, {'clip_gradient': 0.5}]
+    rg_options = [{}, {'rescale_grad': 0.14}, {'rescale_grad': 0.8}]
+    wd_options = [{}, {'wd': 0.03}, {'wd': 0.05}, {'wd': 0.07}]
+    mp_options = [{}, {'multi_precision': False}, {'multi_precision': True}]
+    for dtype in [np.float16, np.float32, np.float64]:
```
[the remark this answers is not in the document]
Comment (D-Roberts) [reply tests/python/unittest/test_optimizer.py]: done

File: tests/python/unittest/test_optimizer.py (hunk 1 above)
Comment (D-Roberts) [reply tests/python/unittest/test_optimizer.py]: done

**Comment by vandanavk** [comment]
@anirudhacharya for another round of review

**Comment by anirudhacharya** [comment]
LGTM

**Comment by vandanavk** [comment]
@mxnet-label-bot update [pr-awaiting-merge]

**Pull request merged**